data/mappings/info_config.json

```diff
@@ -99,4 +101,8 @@
     "USB_MAX_POWER_CONSUMPTION": {"info_key": "usb.max_power", "value_type": "int"},
     "USB_POLLING_INTERVAL_MS": {"info_key": "usb.polling_interval", "value_type": "int"},
     "USB_SUSPEND_WAKEUP_DELAY": {"info_key": "usb.suspend_wakeup_delay", "value_type": "int"},
+
+    # Items we want flagged in lint
+    "NO_ACTION_MACRO": {"info_key": "_ignore.no_action_macro", "deprecated": true},
+    "NO_ACTION_FUNCTION": {"info_key": "_ignore.no_action_function", "removed": true},
```

Comment:
Suggestion: A link to the PR which flags the feature as deprecated or removes it could be added here. So when the lint triggers it provides a context and further information on why the this feature was flagged/removed e.g.:

```
xiudi/xd84: IGNORE_MOD_TAP_INTERRUPT in config.h has been removed and is no longer valid. See https://github.com/qmk/qmk_firmware/pull/15741 for further details.
```

Reply:
@zvecr friendly ping on this proposal.

Further idea for a deprecation workflow of a feature would be:

1. Open a PR which targets `master` which introduces the deprecation warning along with an explanation (in the PR description) why it is deprecated. This is done to announce the deprecation gracefully.
  - e.g. `⚠ whale/sk/v3: SPLIT_TRANSPORT in rules.mk is deprecated and will be removed at a later date. See https://github.com/qmk/qmk_firmware/pull/xxx for further details.`
2. Open a PR which targets `develop` which removes the feature for good and also changes `removed` to true. This breaking change is then introduced with the next breaking change.
  - e.g. `⚠ whale/sk/v3: SPLIT_TRANSPORT in rules.mk has been removed and is no longer valid. See https://github.com/qmk/qmk_firmware/pull/xxx for further details.`

What do you think?

Reply:
Was a bit busy therefore the late answer.

> you don't know the PR number when you write the code

Yes that is a bit annoying, but often we amend to already opened PRs at this point we also know the number. This could also be handled directly with self-suggestions on Github that replace a possible placeholder url.

> PRs can sometimes be too technical for those who consume the info or not provide clear instructions on mitigation

A tldr like reasoning why a feature was removed should be visible at the top of the PR. I think this can often by expressed in one or two sentences without going into the technical detail. Like in the case of the terminal feature #17258 For more sophisticated changes like #15741 there should be a clear instruction in the PR, docs and Release notes on how to mitigate the changes.

> you might not know what breaking changes its going to roll up into so cannot link there

You mean breaking changes document? That would be a problem yes, therefore I would stick to the PR url it self as it is self-contained.

> As for the process, i could see step 1 being done via develop so deprecation happens at a known cadence. In the short term we might have to throw a bunch in to catch up with the current state of play.

Agreed.

The problem I see with not giving a "shortcut" to public information via a warning / error is that people end up opening issues (which are public) or just ask on Discord which is a walled garden and not meant for keeping information accessible.

Reply:
Now we are requiring linked PRs to have a given structure, which they rarely do, and summarise information for non technical users while also potentially going way into the technical details for the change. All while not presenting itself as a wall of text that will get ignored. Users will ask on Discord anyway.

Just seems like more stuff for us to manage, making PRs more of a pain.

Reply:
I'm still not convinced that this is a bad approach. As we are unlikely to converge in this debate I'm resolving this thread in order to not block the merge any further.